paimon-core/src/main/java/org/apache/paimon/utils/BranchManager.java

```diff
@@ -36,6 +36,7 @@ public class BranchManager {
     private static final Logger LOG = LoggerFactory.getLogger(BranchManager.class);
 
     public static final String BRANCH_PREFIX = "branch-";
+    public static final String MAIN_BRANCH = "main";
```

Comment:
What's this `MAIN_BRANCH` used for?

Reply:
Method changes are similar to the following figure:
![image](https://github.com/apache/incubator-paimon/assets/45089228/c26b5035-a38f-4a10-8920-89e5855309be)
By default, we treat it as the main branch. For compatibility with the previous design, branchName = "main" is called if the method is called with no arguments.

Reply:
Rename the field name to DEFAULT_MAIN_BRANCH and if paimon uses "main" as default main, you need to add validation for branch name when users create branches

Reply:
Thanks for your review. I have modified the code according to your suggestion, please review it again.